GiEnJul/Controllers/AdminController.cs

```diff
@@ -47,25 +49,27 @@ public async Task<IEnumerable<Giver>> GetGiversAsync()
             // return _mapper.Map<List<Models.Giver>>(await _giverRepository.GetAllAsync()).OrderBy(x => x.FullName).ToList();
         }
         [HttpGet("recipients")]
-        public async Task<List<Recipient>> GetRecipientsAsync() {
+        public async Task<List<Recipient>> GetRecipientsAsync()
+        {
             var recipients = await _recipientRepository.GetAllAsModelAsync();
             foreach (var recipient in recipients)
             {
                 var familyMembers = await _personRepository.GetAllByRecipientId(recipient.RowKey);
-                recipient.FamilyMembers = familyMembers; 
+                recipient.FamilyMembers = familyMembers;
             }
             return recipients;
         }
         [HttpPost]
-        public async Task<ActionResult> PostConnectionAsyc(string giverRowKey, string recipientRowKey, string partitonKey) {
-            var recipient = await _recipientRepository.GetRecipientAsync(partitonKey, recipientRowKey);
+        public async Task<ActionResult> PostConnectionAsyc([FromBody] PostConnectionDto connectionDto)
+        {
+            var giver = await _giverRepository.GetGiverAsync(connectionDto.GiverPartitionKey, connectionDto.GiverRowKey);
+            var recipient = await _recipientRepository.GetRecipientAsync(connectionDto.RecipientPartitionKey, connectionDto.RecipientRowKey);
             recipient.FamilyMembers = await _personRepository.GetAllByRecipientId(recipient.RowKey);
-            var giver = await _giverRepository.GetGiverAsync(partitonKey, giverRowKey);
             var connection = await _connectionRepository.InsertOrReplaceAsync(giver, recipient);
             giver.IsSuggestedMatch = true;
-            giver.MatchedRecipient = recipientRowKey;
+            giver.MatchedRecipient = connectionDto.RecipientRowKey;
```

Comment:
Could this maybe be done with automapper?

Reply:
The connectionDto is not a model input, it is only four strings so therefore it can not be mapped